src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/delete_categories_modal.html.twig

```diff
@@ -46,6 +46,9 @@
       <div class="d-none">
         {{ form_widget(deleteCategoriesForm.categories_to_delete) }}
       </div>
+      <div class="d-none">
+        {{ form_widget(deleteCategoriesForm.categories_to_delete_parent, { 'attr': {'value': app.request.get('categoryId') } }) }}
```

Comment:
You should inject this data in the controller when creating the form rather than in the template (in `PrestaShopBundle\Controller\Admin\Sell\Catalog\CategoryController::indexAction`)

Reply:
Hi @jolelievre,
Thanks foor feedback !
Could you explain why ? Because in the other templates, it is injected as well
See : https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/cover_image.html.twig#L35

Thanks

Reply:
Yes that's exactly what I thought (only prefer short syntax for array `[]` or it will fail the code style checker)

The reason I advise this is mainly because we use a Symfony form generated from the backoffice, so the whole form (including its default data) should be created on the server side (in the controller) and the template (the view) is only responsible for displaying/rendering it
But it shouldn't inject data in it at the last moment Even the code/hack you needed to make it work is a hint it's not supposed to work that way (inject via attr => value used by the `form_widget` macro)

The other example you refer to is different because:
- the form is completely generated in the template, so no Symfony form in which you could properly inject the data
- it's actually not the form data that is modified, it's a form attribute used in js in which we inject a generate url

Using `app.request.get(*)` in the template might be useful and sometimes necessary, no problem with that (also one should be careful with some potential cache issues) However we are trying to clean all the forms in order to lighten the template as much as possible (see this PR as an example of what we aim https://github.com/PrestaShop/PrestaShop/pull/17558)

And to do that we need clean form types, correctly set in the controller so that you wouldn't even need to modify the template, the idea is that modifying the form type alone will be enough to update the page

I hope this (long) explanation makes sense ^^

Reply:
Oh and I'm not 100% sure what you proposed was never done and can't be found in the core code if you search for it ^^ But it still wouldn't be a good practice

That said, thanks a lot for your contribution in any case 😉

Reply:
It seems legitimate to me. Done !
Thank you @jolelievre